### PR TITLE
Initial release of Soul Reaper

### PR DESCRIPTION
**Pull request by nubspixel**
* **Addressed Issue(s)**: Implements 3CeAM version of Soul Reaper

* **Server Mode**: Both (?)

* **Description of Pull Request**: 
Inspired by the pull request for Star Emperor (/pull/3893), I decided to do the same for Soul Reaper to expedite the process of having both classes in to the main branch. But, I am not confident that I did a good job on the 3CeAM -> rAthena conversion. 

Did a massive mistake in the older pull request (https://github.com/rathena/rathena/pull/3930).

### CODE REVIEW

File: src/map/status.cpp [hunk 1]
```diff
@@ -12247,6 +12341,16 @@ int status_change_end_(struct block_list* bl, enum sc_type type, int tid, const
 			}
 			break;
 
+		case SC_SOULUNITY:
+			{
+				struct block_list *d_bl = map_id2bl(sce->val1);
+				if( d_bl ) {
+					if( d_bl->type == BL_PC )
```
Comment (aleos89) [inline src/map/status.cpp]: You could move this into one check:
```suggestion
					if( d_bl && d_bl->type == BL_PC )
```

**Comment by Busbaty** [comment]
Hello, you make this changes in clif.cpp, but this broken Sura spirit balls effect,

in this part i return every changes 0x1d0 to 0x1e1

static void clif_spiritball_single(int fd, struct map_session_data *sd)
{
WFIFOHEAD(fd, packet_len(0x1d0));
	WFIFOW(fd,0)=0x1d0;
	WFIFOL(fd,2)=sd->bl.id;
	WFIFOW(fd,6)=sd->spiritball;
	WFIFOSET(fd, packet_len(0x1d0));

and this part i change int cmd = 0x1e1; to int cmd = 0x1d0;, i think the rest its not more necessary, but i don't understand this so much, 

int cmd = 0x1e1;
	if (sd && ((sd->class_&MAPID_THIRDMASK) == MAPID_SOUL_REAPER || (sd->class_&MAPID_THIRDMASK) == MAPID_BABY_SOUL_REAPER))
		cmd = 0x1d0;

    WBUFW(buf, 0) = cmd;
    WBUFL(buf, 2) = bl->id;
	WBUFW(buf, 6) = 0; //init to 0
    switch(bl->type){
        case BL_PC: WBUFW(buf, 6) = sd->spiritball; break;
        case BL_HOM: WBUFW(buf, 6) = hd->homunculus.spiritball; break;
    }
    clif_send(buf, packet_len(cmd), bl, AREA);

after that Sura and Soul Reaper still works very nice, i think this change it's not necessary, or just make wrong.

File: src/map/battle.cpp [hunk 2]
```diff
@@ -1216,6 +1216,12 @@ int64 battle_calc_damage(struct block_list *src,struct block_list *bl,struct Dam
 			return 0;
 		}
 
+		if((sce=sc->data[SC_MEIKYOUSISUI]) && rand()%100 < sce->val2)
+		{// Animation is unofficial, but it allows players to know when the nullify chance was successful. [Rytech]
+			clif_specialeffect(bl, 462, AREA);
+			return 0;
+		}
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/battle.cpp]: Removed it completely. Not even sure why/when I copied that over from 3ceam.

File: src/map/clif.cpp [hunk 3]
```diff
@@ -1355,12 +1355,12 @@ void clif_class_change_target(struct block_list *bl,int class_,int type, enum se
 /// 01d0 <id>.L <amount>.W (ZC_SPIRITS)
 /// 01e1 <id>.L <amount>.W (ZC_SPIRITS2)
 static void clif_spiritball_single(int fd, struct map_session_data *sd)
-{
-	WFIFOHEAD(fd, packet_len(0x1e1));
-	WFIFOW(fd,0)=0x1e1;
+{
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/clif.cpp]: done.

File: src/map/clif.cpp [hunk 4]
```diff
@@ -1355,12 +1355,12 @@ void clif_class_change_target(struct block_list *bl,int class_,int type, enum se
...
-{
-	WFIFOHEAD(fd, packet_len(0x1e1));
-	WFIFOW(fd,0)=0x1e1;
+{	
+	WFIFOHEAD(fd, packet_len(0x1d0));
+	WFIFOW(fd,0)=0x1d0;
 	WFIFOL(fd,2)=sd->bl.id;
 	WFIFOW(fd,6)=sd->spiritball;
-	WFIFOSET(fd, packet_len(0x1e1));
+	WFIFOSET(fd, packet_len(0x1d0));
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/clif.cpp]: I thought I reverted it back.  Must have done that on another repo.

File: src/map/map.hpp [hunk 5]
```diff
@@ -184,8 +184,8 @@ enum e_mapid {
 	MAPID_MINSTRELWANDERER,
 	MAPID_SURA,
 	MAPID_GENETIC,
-	MAPID_SHADOW_CHASER,
-	MAPID_SOUL_REAPER,
+	MAPID_SHADOW_CHASER,
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/map.hpp]: done.

File: src/map/skill.cpp [hunk 6]
```diff
@@ -16116,6 +16367,7 @@ void skill_consume_requirement(struct map_session_data *sd, uint16 skill_id, uin
 
 	if( type&1 ) {
 		switch( skill_id ) {
+
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/skill.cpp]: done.

File: src/map/skill.cpp [hunk 7]
```diff
@@ -16131,9 +16383,10 @@ void skill_consume_requirement(struct map_session_data *sd, uint16 skill_id, uin
 		}
 		if(require.hp || require.sp)
 			status_zap(&sd->bl, require.hp, require.sp);
-
-		if(require.spiritball > 0)
+		
+		if(require.spiritball > 0) {
 			pc_delspiritball(sd,require.spiritball,0);
+		}
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/skill.cpp]: done.

File: src/map/status.cpp [hunk 8]
```diff
@@ -11274,6 +11321,53 @@ int status_change_start(struct block_list* src, struct block_list* bl,enum sc_ty
 			val2 = 30; // SP Recovery rate bonus
 			break;
 
+		case SC_SOULGOLEM: 
+			#ifdef RENEWAL
+				val2 = 60 * val1;// DEF Increase
+			#else
+				val2 = 60 * val1/10;// DEF Increase
+			#endif
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/status.cpp]: done.

File: src/map/skill.cpp [hunk 9]
```diff
@@ -16131,9 +16383,10 @@ void skill_consume_requirement(struct map_session_data *sd, uint16 skill_id, uin
 		}
 		if(require.hp || require.sp)
 			status_zap(&sd->bl, require.hp, require.sp);
-
-		if(require.spiritball > 0)
+
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/skill.cpp]: done.

File: src/map/status.cpp (hunk 1 above)
Comment (nubspixel) [reply src/map/status.cpp]: done.

File: src/map/map.hpp [hunk 10]
```diff
@@ -215,8 +215,8 @@ enum e_mapid {
 	MAPID_BABY_MINSTRELWANDERER,
 	MAPID_BABY_SURA,
 	MAPID_BABY_GENETIC,
-	MAPID_BABY_CHASER,
-	MAPID_BABY_SOUL_REAPER,
+	MAPID_BABY_CHASER,
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/map.hpp]: done.

**Comment by nubspixel** [comment]
@Busbaty I thought i already reverted that. I might have confused about it with another project. The changes on clif.cpp have been reverted back to original. It was a mistake of mine when i was testing.

File: src/map/skill.cpp [hunk 11]
```diff
@@ -8753,8 +8865,20 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 	case SL_STAR:
 	case SL_SUPERNOVICE:
 	case SL_WIZARD:
+	case SL_DEATHKNIGHT: // ?
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/skill.cpp]: done.

File: src/map/skill.cpp [hunk 12]
```diff
@@ -15885,7 +16136,7 @@ bool skill_check_condition_castbegin(struct map_session_data* sd, uint16 skill_i
 	}
 
 	if ((require.spiritball > 0 && sd->spiritball < require.spiritball) ||
-		(require.spiritball == -1 && sd->spiritball < 1)) {
+		(require.spiritball == -1 && sd->spiritball < 1 )) {
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/skill.cpp]: done.

File: src/map/skill.cpp [hunk 13]
```diff
@@ -7276,25 +7347,62 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
...
 			}
 		}
-		clif_skill_nodamage(src,bl,skill_id,skill_lv,
-			sc_start(src,bl,type,100,skill_lv,skill_get_time(skill_id, skill_lv)));
+
+		if ( skill_id == SP_KAUTE )
+		{
+			int hp, sp;
+			hp = sstatus->max_hp * (10 + 2 * skill_lv) / 100;
+			sp = tstatus->max_sp * (10 + 2 * skill_lv) / 100;
```
[the remark this answers is not in the document]
Comment (nubspixel) [reply src/map/skill.cpp]: done.

**Comment by sianromantic** [comment]
@nubspixel the skill([Evil spirit's Curse](https://www.divine-pride.net/database/skill/2601)) not work.
If i use [Evil spirit's Curse](https://www.divine-pride.net/database/skill/2601) first then use [Curse explosion](https://www.divine-pride.net/database/skill/2600) , damage not change.

**Comment by nubspixel** [comment]
@sianromantic the target must be Cursed for the damage buff to work.

**Comment by rofasindo** [comment]
@nubspixel did the skill [Soul Explosion](https://www.divine-pride.net/database/skill/2610) work? I have tested and always miss.

**Comment by nubspixel** [comment]
@rofasindo what soul did you use on the target?

**Comment by rofasindo** [comment]
@nubspixel I have tried use every soul on target (Soul Falcon, Soul Golem, Soul Fairy, Soul Shadow), then I use Soul Explosion then nothing happen on the target. I just see Miss

**Comment by nubspixel** [comment]
@rofasindo yeah. I must have forgotten to add the damage calculation for that skill.

On an unrelated note, seems like soul ball requires 0x1d0 instead of 0x1e1 in clif_spiritball_single. Using 0x1d0 is fine on the Soul Repear player. However, another client will see both the animation/effects of soul ball and spirit ball. Using 0x1e1 seems to mitagate this issue. Thus, a condition to only use 0x1d0 for Soul Repear class is added.

**Comment by Badarosk0** [comment]
I noticed that the map server is randomly freezing and knocking over. I constantly get the error in the log:
```
[Error]: WFIFOSET: Maximum write buffer size for client connection 35 exceeded, most likely caused by packet 0x01d0 (len=8, ip=187.14.196.235).
[Error]: WFIFOSET: Maximum write buffer size for client connection 35 exceeded, most likely caused by packet 0x01d0 (len=8, ip=187.14.196.235).
```
Any idea how to solve it? @nubspixel

**Comment by nubspixel** [comment]
@Badarosk0 I am sorry, i can't really help with packets issue. I don't really understand how packets works. And so far, i don't have crashes because of that packet.

File: src/map/battle.cpp [hunk 14]
```diff
@@ -4253,7 +4264,8 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
 		case KO_SETSUDAN:
 			skillratio += 100 * (skill_lv - 1);
 			RE_LVL_DMOD(100);
-			if(tsc && tsc->data[SC_SPIRIT])
+			// Bonus damage added when target is soul linked.
+			if(tsc && ( tsc->data[SC_SPIRIT] || tsc->data[SC_SOULGOLEM] || tsc->data[SC_SOULSHADOW] || tsc->data[SC_SOULFALCON] || tsc->data[SC_SOULFAIRY]))
```
Comment (mrjnumber1) [inline src/map/battle.cpp]: this needs to check which tsc->data[] is being used!

**Comment by Badarosk0** [comment]
Skill Shadow's Soul is increasing ASPD to a value much greater than 3 units. @nubspixel

File: src/map/battle.cpp (hunk 14 above)
Comment (nubspixel) [reply src/map/battle.cpp]: Not sure if this is required. I never see any other sc->data being checked.

**Comment by nubspixel** [comment]
@Badarosk0 fixed the aspd issue.

File: src/map/battle.cpp (hunk 14 above)
Comment (mrjnumber1) [reply src/map/battle.cpp]: i dont follow your reasoning.  if all that matters is the level of SC_SPIRIT the player gets, then what's hte point of checking the reaper statuses? they all dispel SC_SPIRIT.

**Comment by Badarosk0** [comment]
Crash map server @nubspixel :
```
Program received signal SIGSEGV, Segmentation fault.
0x0000000000612537 in skill_check_condition_castbegin (sd=0x78715a0, skill_id=469, skill_lv=10) at skill.cpp:16040
16040				if(!(sc && sc->data[SC_SMA] || sc->data[SC_USE_SKILL_SP_SHA]))
Missing separate debuginfos, use: debuginfo-install glibc-2.12-1.209.el6_9.2.x86_64 keyutils-libs-1.4-5.el6.x86_64 krb5-libs-1.10.3-65.el6.x86_64 libcom_err-1.41.12-23.el6.x86_64 libgcc-4.4.7-18.el6.x86_64 libselinux-2.0.94-7.el6.x86_64 libstdc++-4.4.7-18.el6.x86_64 mysql-libs-5.1.73-8.el6_8.x86_64 nss-softokn-freebl-3.14.3-23.3.el6_8.x86_64 openssl-1.0.1e-57.el6.x86_64 pcre-7.8-7.el6.x86_64 zlib-1.2.3-29.el6.x86_64
(gdb) 
[K(gdb) bt full
#0  0x0000000000612537 in skill_check_condition_castbegin (sd=0x78715a0, skill_id=469, skill_lv=10) at skill.cpp:16040
        status = 0x7871b04
        sc = 0x0
        require = {hp = 0, mhp = 0, sp = 41, hp_rate = 0, sp_rate = 0, zeny = 0, weapon = 0, ammo = 0, ammo_qty = 0 '\000', state = 0 '\000', spiritball = 0 '\000', 
          itemid = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0}, amount = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0}, eqItem = 0x0, status = 0x0, status_count = 0 '\000', 
          eqItem_count = 0 '\000'}
        i = <value optimized out>
        inf2 = <value optimized out>
        inf3 = <value optimized out>
        __FUNCTION__ = "skill_check_condition_castbegin"
#1  0x0000000000680b82 in unit_skilluse_id2 (src=0x78715a0, target_id=110213162, skill_id=Unhandled dwarf expression opcode 0xf3
) at unit.cpp:1709
        ud = 0x78715c0
        tstatus = 0x7fffd8ddbec0
        sc = <value optimized out>
        sd = 0x78715a0
        target = <value optimized out>
        tick = 2129686761
        combo = 0
        range = <value optimized out>
        inf = 192 '\300'
        inf2 = 3638409684
        __FUNCTION__ = "unit_skilluse_id2"
#2  0x00000000004cb4b8 in clif_parse (fd=Unhandled dwarf expression opcode 0xf3
) at clif.cpp:21763
        cmd = 1080
        packet_len = 10
        sd = 0x78715a0
        pnum = 0
#3  0x00000000006880fd in do_sockets (next=Unhandled dwarf expression opcode 0xf3
) at socket.cpp:1407
        rfd = {fds_bits = {33554432, 0 <repeats 15 times>}}
        timeout = {tv_sec = 0, tv_usec = 15717}
        ret = 0
        i = 25
#4  0x000000000046dda5 in main (argc=1, argv=0x7fffffffde28) at core.cpp:369
        next = <value optimized out>
```

**Comment by nubspixel** [comment]
> Crash map server @nubspixel :
> 
> ```
> Program received signal SIGSEGV, Segmentation fault.
> 0x0000000000612537 in skill_check_condition_castbegin (sd=0x78715a0, skill_id=469, skill_lv=10) at skill.cpp:16040
> 16040				if(!(sc && sc->data[SC_SMA] || sc->data[SC_USE_SKILL_SP_SHA]))
> Missing separate debuginfos, use: debuginfo-install glibc-2.12-1.209.el6_9.2.x86_64 keyutils-libs-1.4-5.el6.x86_64 krb5-libs-1.10.3-65.el6.x86_64 libcom_err-1.41.12-23.el6.x86_64 libgcc-4.4.7-18.el6.x86_64 libselinux-2.0.94-7.el6.x86_64 libstdc++-4.4.7-18.el6.x86_64 mysql-libs-5.1.73-8.el6_8.x86_64 nss-softokn-freebl-3.14.3-23.3.el6_8.x86_64 openssl-1.0.1e-57.el6.x86_64 pcre-7.8-7.el6.x86_64 zlib-1.2.3-29.el6.x86_64
> (gdb) 
> �[K(gdb) bt full
> #0  0x0000000000612537 in skill_check_condition_castbegin (sd=0x78715a0, skill_id=469, skill_lv=10) at skill.cpp:16040
>         status = 0x7871b04
>         sc = 0x0
>         require = {hp = 0, mhp = 0, sp = 41, hp_rate = 0, sp_rate = 0, zeny = 0, weapon = 0, ammo = 0, ammo_qty = 0 '\000', state = 0 '\000', spiritball = 0 '\000', 
>           itemid = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0}, amount = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0}, eqItem = 0x0, status = 0x0, status_count = 0 '\000', 
>           eqItem_count = 0 '\000'}
>         i = <value optimized out>
>         inf2 = <value optimized out>
>         inf3 = <value optimized out>
>         __FUNCTION__ = "skill_check_condition_castbegin"
> #1  0x0000000000680b82 in unit_skilluse_id2 (src=0x78715a0, target_id=110213162, skill_id=Unhandled dwarf expression opcode 0xf3
> ) at unit.cpp:1709
>         ud = 0x78715c0
>         tstatus = 0x7fffd8ddbec0
>         sc = <value optimized out>
>         sd = 0x78715a0
>         target = <value optimized out>
>         tick = 2129686761
>         combo = 0
>         range = <value optimized out>
>         inf = 192 '\300'
>         inf2 = 3638409684
>         __FUNCTION__ = "unit_skilluse_id2"
> #2  0x00000000004cb4b8 in clif_parse (fd=Unhandled dwarf expression opcode 0xf3
> ) at clif.cpp:21763
>         cmd = 1080
>         packet_len = 10
>         sd = 0x78715a0
>         pnum = 0
> #3  0x00000000006880fd in do_sockets (next=Unhandled dwarf expression opcode 0xf3
> ) at socket.cpp:1407
>         rfd = {fds_bits = {33554432, 0 <repeats 15 times>}}
>         timeout = {tv_sec = 0, tv_usec = 15717}
>         ret = 0
>         i = 25
> #4  0x000000000046dda5 in main (argc=1, argv=0x7fffffffde28) at core.cpp:369
>         next = <value optimized out>
> ```

Sorry, but my skills aren't really that good to understand much about gdb output but from my understanding, it does not have anything to do with soul reaper. But then again, I am not sure. Do you know how to reproduce the crash?

**Comment by Badarosk0** [comment]
There is something wrong with the line 
(
if (! (Sc && sc-> data [SC_SMA] || sc-> data [SC_USE_SKILL_SP_SHA])
)
 in skill.cpp. What does this line do? 
Can it be removed?
@nubspixel

**Comment by mrjnumber1** [comment]
yeah. it's because of a mistaken nullptr issue. the second or is checked even if sc is null. it should be as such:
```if (sc && ! (sc->data [SC_SMA] || sc->data [SC_USE_SKILL_SP_SHA])```
also my line about SC_SOULGOLEM and such above had an off by 1 error. so make sure u have that checking i too!

**Comment by DepTrai92** [comment]
[Error]: WFIFOSET: Maximum write buffer size for client connection 8 exceeded, most likely caused by packet 0x01d0 (len=8, ip=).

**Comment by DepTrai92** [comment]
> [Error]: WFIFOSET: Maximum write buffer size for client connection 8 exceeded, most likely caused by packet 0x01d0 (len=8, ip=).

i have a error,when i login soul reaper.

**Comment by Badarosk0** [comment]
> > [Error]: WFIFOSET: Maximum write buffer size for client connection 8 exceeded, most likely caused by packet 0x01d0 (len=8, ip=).
> 
> i have a error,when i login soul reaper.

Yes, this problem is real. Confirmed.

You need to suspend the Soul Reaper login to stop occurring.

**Comment by DepTrai92** [comment]
@nubspixel  please advise.

**Comment by nubspixel** [comment]
Strange. Why did it start to occurred suddenly? hmm..

**Comment by attackjom** [comment]
> Strange. Why did it start to occurred suddenly? hmm..

Hi @nubspixel I have done a seperate soulball system from spirit ball, would you mind if I want the permission on this repo ?

**Comment by nubspixel** [comment]
@Badarosk0 @DepTrai92 did the two of you increase the MAX_SPIRITBALL (in pc.hpp) and if so, by how much?

**Comment by nubspixel** [comment]
> > Strange. Why did it start to occurred suddenly? hmm..
> 
> Hi @nubspixel I have done a seperate soulball system from spirit ball, would you mind if I want the permission on this repo ?

I don't see a good reason to separate them as it is using the same packet (relatively). you can't actually have both spiritball and soulball at the same time.

**Comment by Badarosk0** [comment]
> Strange. Why did it start to occurred suddenly? hmm..

This has been the case since it was created. This is somewhat random, but if the crash occurs occurs when the person login in with the soul reaper. 
It may be that the person has always played with soul reaper without problems and at some point the problem occurs and just the person entering with this char that is going to bring down the server.

> @Badarosk0 @DepTrai92 did the two of you increase the MAX_SPIRITBALL (in pc.hpp) and if so, by how much?

My MAX_SPIRITBALL (in pc.hpp) is 20.

**Comment by attackjom** [comment]
share my seperated soulball system
https://gist.github.com/attackjom/ad8696748536b0c2deece2a5378bc3a1#file-soulball-diff

**Comment by Badarosk0** [comment]
Was this version used by Rytech? I had used it separately before and the problem also occurred. Resolved your issue with this change? @attackjom

**Comment by attackjom** [comment]
> Was this version used by Rytech? I had used it separately before and the problem also occurred. Resolved your issue with this change? @attackjom

I have use in my private server  around 25 person in server and its not happen any error of WFIoffset in my map-server for more than month

**Comment by DepTrai92** [comment]
Hi,i have try recode again.is dont have the error anymore

**Comment by besttide155** [comment]
The skill "Soul Collect" doesn't work properly. The icon of this skill displays negative values depending on skill's level. You can hover into the icon to see it. And because of it, the collected souls decreases to 1 after I relogging no matter how many soul I held.

image

**Comment by besttide155** [comment]
Hi again, when i cast Fairy's Soul, the magical damage inflicted does increase but the bonus MATK is not added in Stats Dialog (Alt+A in game). Other skills (Shadow, Golem & Falcon) give the bonus stat amount properly.

**Comment by Anjuts** [comment]
Soul Unity is crashing. I forgot to bt full.

File: db/re/skill_db.txt [hunk 15]
```diff
@@ -1329,24 +1329,23 @@
...
+2596,9,6,16,0,0x1,0,5,1,yes,0,0,0,magic,0,0x0,	SP_SOULGOLEM,Soul Golem
+2597,9,6,16,0,0x1,0,5,1,yes,0,0,0,magic,0,0x0,	SP_SOULSHADOW,Soul Shadow
+2598,9,6,16,0,0x1,0,5,1,yes,0,0,0,magic,0,0x0,	SP_SOULFALCON,Soul Falcon
+2599,9,6,16,0,0x1,0,5,1,yes,0,0,0,magic,0,0x0,	SP_SOULFAIRY,Soul Fairy
+2600,9,8,1,7,0x2,3,5,-7,yes,0,0,0,magic,0,0x0,	SP_CURSEEXPLOSION,Curse Explosion
+2601,9,6,1,0,0x3,3,5,1,yes,0,0,0,magic,0,0x0,	SP_SOULCURSE,Soul Curse
+2602,9,6,1,-2,0,0,5,1,yes,0,0,0,magic,0,0x0,	SP_SPA,Espa
+2603,9,6,1,-2,0x2,1:1:2:2:2,5,1,yes,0,0,0,magic,0,0x0,	SP_SHA,Esha
+2604,9,8,1,-2,0x2,1:1:1:2:2:2:3,7,-5,yes,0,0,0,magic,0,0x0,	SP_SWHOO,Eswhoo
+2605,11,6,4,0,0x3,1:1:2:2:3:3:4,7,1,yes,0,0,0,magic,0,0x0,	SP_SOULUNITY,Soul Unity
```
Comment (Questune09) [inline db/re/skill_db.txt]: Server crashed after casting soul unity

**Comment by aleos89** [comment]
Moved to #4386 so the rAthena staff can manage it. Thank you for your contributions!

**Pull request closed without merging**